pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```diff
@@ -0,0 +1,221 @@
+         *      If this is null or empty, the default JVM trust store is used. In this demo, we use a provided
+         *      "cert.pem" as the CA certificate, which is also provided on the server-side. If the cluster uses a
+         *      different CA (which it should), use that CA's certificate as the truststore instead.
+         *
+         *    - Also, the client config below disables host name verification. If the cluster's server certificates
+         *      have DNS names / IP addresses of the servers specified in them, you may turn this on. In a production
+         *      deployment, it is recommended to keep this on.
+         *
+         * Note about setting the client config for auth:
+         *    - The client config below is configured with an object of DefaultCredentials class. The user name
+         *      and password arguments passed to the object represent the credentials used for authentication
+         *      and authorization. The assumption we are making here is that the username is valid on the server,
+         *      the password is correct and the username has all the permissions necessary for performing the
+         *      subsequent operations.
+         */
+        clientConfig = ClientConfig.builder()
+                .controllerURI(this.controllerURI) // "tls://localhost:9090"
+
+                // TLS-related client-side configuration
+                .trustStore(this.truststorePath)
+                .validateHostName(this.validateHostName)
+
+                // Auth-related client-side configuration
+                .credentials(new DefaultCredentials(this.password, this.username))
+                .build();
+
+        System.out.println("Done creating a client config.");
+
+        // Everything below depicts the usual flow of reading events. All client-side security configuration is
+        // done through the ClientConfig object as shown above.
+
+        try (BatchClientFactory batchClient = BatchClientFactory.withScope(scope, clientConfig)) {
+            System.out.println("Done creating batchClient for " + scope + "/" + stream);
+            ArrayList<SegmentRange> segments = Lists.newArrayList(batchClient.getSegments(Stream.of(scope, stream), StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+            readFromSegments(batchClient, segments);
+            System.out.println("Done reading " + String.valueOf(segments.size()) + " segments.");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        System.err.println("All done with reading! Exiting...");
+    }
+
+    public static void main(String[] args) throws ReinitializationRequiredException {
+        Options options = getOptions();
+
+        CommandLine cmd = null;
+        try {
+            cmd = parseCommandLineArgs(options, args);
+        } catch (ParseException e) {
+            System.out.format("%s.%n", e.getMessage());
+            final HelpFormatter formatter = new HelpFormatter();
```

Comment:
What is this for? Also `HelloWorldReader` does not seem related to this sample, which is a batch reader. Remove unnecessary stuff if possible to keep this simple.